File: hub/echo-agent/bridge.js
#!/usr/bin/env node
'use strict';
/**
 * Minimal AgentProc echo agent (Node.js, wire 0.3).
 *
 * Reads the {"type":"turn",...} object from stdin and writes the message back
 * as a single {"type":"text"} event. No external dependencies, no AI calls.
 * Use this to verify your messaging bridge speaks the protocol correctly.
 */

const readline = require('node:readline');

const rl = readline.createInterface({ input: process.stdin });
rl.on('line', (line) => {
  let turn = {};
  try { turn = JSON.parse(line); } catch { /* empty turn */ }
  const message = (typeof turn.message === 'string') ? turn.message : '';
  process.stdout.write(JSON.stringify({ type: 'text', text: `You said: ${message}` }) + '\n');
  process.exit(0);
});
rl.on('close', () => {
  // No turn line at all — echo an empty message.
  process.stdout.write(JSON.stringify({ type: 'text', text: 'You said: ' }) + '\n');
  process.exit(0);
});

File: hub/echo-agent/bridge.py
#!/usr/bin/env python3
"""
Minimal AgentProc echo agent (Python, wire 0.3).

Reads the {"type":"turn",...} object from stdin and writes the message back
as a single {"type":"text"} event. No external dependencies, no AI calls.
Use this to verify your messaging bridge speaks the protocol correctly.
"""

import json
import sys


def main() -> int:
    try:
        line = sys.stdin.readline()
        turn = json.loads(line.rstrip("\r\n")) if line else {}
    except (json.JSONDecodeError, OSError):
        turn = {}
    message = turn.get("message") if isinstance(turn.get("message"), str) else ""
    sys.stdout.write(
        json.dumps({"type": "text", "text": f"You said: {message}"}, ensure_ascii=False, separators=(",", ":")) + "\n"
    )
    sys.stdout.flush()
    return 0


if __name__ == "__main__":
    sys.exit(main())
